packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/sync/SyncSession.kt

```diff
@@ -107,6 +111,32 @@ public interface SyncSession {
      */
     public fun resume()
 
+    /**
+     * Create a [Flow] of [Progress]-events that track either downloads or uploads done by the [SyncSession].
+     *
+     * This is only an indicator of transferring of data and an [Progress]-event with
+     * [Progress.isTransferComplete] being `true` does not guarantee that the data is already
+     * visible in the realm. To wait for data being integrated and visible use
+     * [downloadAllServerChanges]/[uploadAllLocalChanges].
+     *
+     * If the flow is created with [ProgressMode.CURRENT_CHANGES] the [Progress] will
+     * only ever increase and will complete once `Progress.isTransferComplete = true`.
+     *
+     * If the flow is created with [ProgressMode.INDEFINITELY] the [Progress] can both
+     * increase and decrease since more changes might be added while the flow is still active. This
+     * means that it is possible for one [Progress] instance to report
+     * `isTransferComplete = true` and subsequent instances to report `isTransferComplete = false`.
+     *
+     * The flow has an internal buffer of [Channel.BUFFERED] but if the consumer fails to consume the
```

Comment:
Should this be a `@throws` instead?

Reply:
Definitely not, since it is not the `progress` calls that throws. But I don't know how it should be documented then. I was trying to find similar cases but didn't find any. I am open to suggestions.

Reply:
I was unsure myself. It is fine just keeping this as it is.